core/kernel/dt_driver.c

```diff
@@ -244,6 +246,28 @@ static void *device_from_provider_prop(struct dt_driver_provider *prv,
 	return device;
 }
 
+void *dt_driver_device_from_parent(const void *fdt, int nodeoffset,
+				   enum dt_driver_type type, TEE_Result *res)
+{
+	int parent = -1;
+	struct dt_driver_provider *prv = NULL;
+
+	parent = fdt_parent_offset(fdt, nodeoffset);
+	if (parent < 0) {
+		*res =  TEE_ERROR_GENERIC;
+		return NULL;
+	}
+
```

Comment:
should this check that `fdt` matches the FDT use when provider was registered?
I think `assert(fdt == get_secure_dt());` would be enough for now, seen _dt_driver.c_ current implementation.

Reply:
Acked